### PR TITLE
Pass NodeTemplate to MachineClass after reading from Worker resource

### PR DESCRIPTION
**Pull request by himanshu-kun**
**How to categorize this PR?**

/area scalability
/kind enhancement
/platform gcp

**What this PR does / why we need it**:
This PR enables extension GCP 
* to read [nodeTemplate](https://github.com/gardener/gardener/blob/a408d6d9720c170b377c6476b59a5ace5fff7a6b/pkg/apis/extensions/v1alpha1/types_worker.go#L152) from worker resource and pass it onto machineClass with extra fields like InstanceType,region and zone. This is required for generic-scale-from-zero.
* removes the validation logic introduced earlier because of absence of scale-from-zero support for GCP.
* allows users to provide runtime resources in `providerConfig` in shoot.yaml and then prioritizing it over cloudprofile resources while filling it in machineClass 

**Which issue(s) this PR fixes**:
Fixes #
A step in [gardener/autoscaler#27](https://github.com/gardener/autoscaler/issues/27)

**Special notes for your reviewer**:

**Release note**:

```feature user
extension gcp now can read nodeTemplate from worker resource and fill it in machineClass. It prioritizes nodeTemplate provided by user in `providerConfig` (under `worker` section in shoot yaml) over worker resource's nodeTemplate, if user provides it.
```

### CODE REVIEW

**Comment by himanshu-kun** [comment]
cc @AxiomSamarth

File: pkg/controller/worker/machines.go [hunk 1]
```diff
@@ -175,6 +181,7 @@ func (w *workerDelegate) generateMachineConfig(ctx context.Context) error {
 				"disks":              disks,
 				"labels":             gceInstanceLabels,
 				"machineType":        pool.MachineType,
+				"nodeTemplate":       nodeTemplate,
```
Comment (AxiomSamarth) [inline pkg/controller/worker/machines.go:184]: Why not keep things simple and create the whole `nodeTemplate` object right here?

**Review by AxiomSamarth — CHANGES_REQUESTED** [review]

File: pkg/controller/worker/machines.go (hunk 1 above)
Comment (himanshu-kun) [reply pkg/controller/worker/machines.go:184]: sure

**Comment by himanshu-kun** [comment]
cc @vpnachev

**Comment by vpnachev** [comment]
> cc @vpnachev

What is expected from me?

**Comment by himanshu-kun** [comment]
> What is expected from me?

I should have stated the reason, sorry. I needed early review from you on this PR. Pls do whenever you are free. Thanks

**Comment by vpnachev** [comment]
> 

There is an `/invite` command to request a PR review, :wink:

**Comment by himanshu-kun** [comment]
Thanks will remember that:)
/invite @vpnachev

File: docs/usage-as-end-user.md [hunk 2]
```diff
@@ -125,36 +125,6 @@ If you don't want to configure anything for the `cloudControllerManager` simply
 
 ## WorkerConfig
 
-Multiple zones can be configured for a worker group of a GCP Shoot. The minimum number of machines in every worker group should be equal to or greater than the number of zones configured for that worker-group.
```
Comment (vpnachev) [inline docs/usage-as-end-user.md:128]: Why is this documentation removed?

Comment (himanshu-kun) [reply docs/usage-as-end-user.md:128]: With scale from zero enabled on GCP, we won't need to restrict user with `min machine in worker pool >= no of zones" , thats why I removed it.
The [PR](https://github.com/gardener/gardener-extension-provider-gcp/pull/296) which added this validation before had added the mentioned lines in the documentation , so I removed the same.

Comment (vpnachev) [reply docs/usage-as-end-user.md:128]: Thanks! I had to read carefully what is actually removed as I was thinking you are removing a general documentation about how multiple availability zones to be configured. Thanks for clarifying ans sorry for the inconvenience

**Review by dkistner — APPROVED** [review]
/lgtm
/ping @himanshu-kun

**Review by kon-angelo — APPROVED** [review]
/lgtm

**Pull request merged**